jvm/src/main/kotlin/app/cash/trifle/internal/validators/X509CertChainValidator.kt

```diff
@@ -43,7 +44,21 @@ internal class X509CertChainValidator(certAnchor: Certificate, date: Date? = nul
       return false
     }
 
-    PATH_VALIDATOR.validate(X509FACTORY.generateCertPath(x509Certs), pkixParams)
+    try {
+      PATH_VALIDATOR.validate(X509FACTORY.generateCertPath(x509Certs), pkixParams)
+    } catch (e: CertPathValidatorException) {
+      val reason = e.reason
+      // https://docs.oracle.com/javase/8/docs/api/java/security/cert/PKIXReason.html
+      // https://docs.oracle.com/javase/8/docs/api/java/security/cert/CertPathValidatorException.BasicReason.html
+      when (reason) {
+          BasicReason.EXPIRED -> throw ExpiredCertificateException("Expired Trifle certificate", e)
```

Comment:
here we're just re-wrapping the underlying exception then re-throwing. How might this be better than letting the exception escape?

Reply:
We are making it more meaningful for the calling app. We do not expect the calling app to go look at all the exceptions of JCE and letting them know only the exceptions they will care about.

Reply:
hmm what exactly is meaningful? We have a catch all `UnSpecifiedFailureException` to represent the remaining set of enums found in `BasicReason` and `PKIXReason`. As a client how would this be different if the exception handling is dealt:

```
try {
 ...
} catch (e: CertPathValidatorException) {
  when (e.reason) {
     BasicReason.EXPIRED -> ...
     BasicReason.INVALID_SIGNATURE -> ...
     else -> ...
  }
}
```

vs

```
try {
 ...
} catch (e: ExpiredCertificateException) {
  ...
} catch (e: IncorrectSignatureException) {
  ...
} catch (e: NoTrustAnchorException) {
  ...
} catch (e: UnSpecifiedFailureException) {
  ...
}
```